gateway/webhook/webhook.go

```diff
@@ -261,6 +262,41 @@ func (webhook *HandleT) batchRequests(sourceDef string, requestQ chan *webhookT)
 	}
 }
 
+func prepareRequestBody(req *http.Request, includeQueryParams bool, sourceType string, sourceListForParsingParams []string) ([]byte, error) {
+	defer func() {
+		_ = req.Body.Close()
+	}()
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, errors.New(response.RequestBodyReadFailed)
+	}
+
+	if len(body) == 0 {
+		body = []byte("{}") // If body is empty, set it to an empty JSON object
+	} else {
+		// Validate the JSON
+		var temp interface{}
+		if json.Unmarshal(body, &temp) != nil {
+			return nil, errors.New(response.InvalidJSON)
+		}
+	}
```

Comment:
We probalby don't need the else block, since `sjson` would fail if it will be not able to set the `query_parameters`.

Reply:
No, if we send normal string it was not verifying this particular function. Tests were failing. Hence added this block .

Reply:
I mean the objective of the function is not to check whether the extracted payload is proper JSON or not. It's just to extract the payload right. We are checking later in the flow whether the payload is valid JSON or not.

Reply:
I was thinking, we could remove the validation from here since we are already doing it later. Test the behaviour of the function in case of Invalid JSON.

```GO
func TestPrepareRequestBody(t *testing.T) {
	createRequest := func(method, target string, body io.Reader, params map[string]string) *http.Request {
		r := httptest.NewRequest(method, target, body)
		q := r.URL.Query()
		for k, v := range params {
			q.Add(k, v)
		}
		r.URL.RawQuery = q.Encode()
		return r
	}

	testCases := []struct {
		name               string
		req                *http.Request
		includeQueryParams bool
		wantError          bool
		expectedResponse   []byte
	}{
		{
			name:               "Empty request body and no query parameters",
			req:                createRequest(http.MethodPost, "http://example.com", nil, nil),
			includeQueryParams: false,
			expectedResponse:   []byte("{}"),
		},
		{
			name:               "Error reading request body",
			req:                createRequest(http.MethodPost, "http://example.com", iotest.ErrReader(errors.New("some error")), nil),
			includeQueryParams: false,
			wantError:          true,
			expectedResponse:   nil,
		},
		{
			name:               "Valid JSON body with no query parameters",
			req:                createRequest(http.MethodPost, "http://example.com", strings.NewReader(`{"key":"value"}`), nil),
			includeQueryParams: false,
			expectedResponse:   []byte(`{"key":"value"}`),
		},
		{
			name:               "Valid JSON body with query parameters",
			req:                createRequest(http.MethodPost, "http://example.com", strings.NewReader(`{"key1":"value1"}`), map[string]string{"key2": "value2"}),
			includeQueryParams: true,
			expectedResponse:   []byte(`{"key1":"value1","query_parameters":{"key2":["value2"]}}`),
		},
		{
			name:               "Invalid JSON body with query parameters",
			req:                createRequest(http.MethodPost, "http://example.com", strings.NewReader(`{"key1":"value1", "key2":}`), map[string]string{"key3": "value3"}),
			includeQueryParams: true,
			expectedResponse:   []byte(`{"key1":"value1", "key2":,"query_parameters":{"key3":["value3"]}}`),
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			result, err := prepareRequestBody(tc.req, tc.includeQueryParams, "webhook", []string{"webhook"})
			if tc.wantError {
				require.Error(t, err)
				return
			}
			t.Log(string(result))
			require.NoError(t, err)
			require.Equal(t, tc.expectedResponse, result)
		})
	}
}
```

Reply:
got your point. removing the validation from util then